### ISSUE TITLE
Change-Request: `CPM_USE_NAMED_CACHE_DIRECTORIES` should be removed or default to TRUE

### ISSUE THREAD

**Issue by CraigHutchinson** [cpm-cmake/CPM.cmake#408]
# Background 
The `CPM_USE_NAMED_CACHE_DIRECTORIES` option adds an extra folder which contains the name of the package under the SHA folder. This is so instead of showing SHA under `CLion` or `Visual Studio` you get a proper-name.

# Issue
Users need to know that `CPM_USE_NAMED_CACHE_DIRECTORIES` exists when using CPM for the IDE to not show a horrendous list of SHAs!
![image](https://user-images.githubusercontent.com/54269136/190675332-8ea7935c-6ee6-4490-b7ce-3f6cb8d0f72e.png)

# Proposition
I cannot see any reason not to add the Named-Cache directory always and deprecate the option. This both simplifies CPM maintenance and also the end-user experience.

I can see some trivial issues:
- a few extra characters in the path-name length 
- existing CPM caches could be invalidated by the change (TBD)
    - IDEA: Cache folder should have `version` file at root so the cache structure can be cleared to smooth future updates?

**Comment by TheLartians** [comment]
Yeah, as I remember the main reason why we still have the option is to prevent having multiple cache directories build up in parallel depending on the used CPM version of the underlying project. I agree that we should switch to the more compatible format rather sooner or later, though I'm not sure about the best approach.

> IDEA: Cache folder should have version file at root so the cache structure can be cleared to smooth future updates

Not a bad idea, though we should not automatically clear the cache, as some users might have projects with old CPM.cmake versions on the same machine. On the other had, separating the caches into separate directories by version would make it easier for the users to remove one or the other completely.

**Comment by CraigHutchinson** [comment]
> though we should not automatically clear the cache, as some users might have projects with old CPM.cmake versions on the same machine.

That is a really good point. Any change should avoid being destructive in this respect.

 Another potential use-case may be be when the developer no longer has access to the source repo to be able to re-clone  and is relying on the cached checkout persisting. This could be due to needing a local on-site git server etc. Definitely should avoid any unnecessary headaches.
 
 One idea I just had is instead of just a version file is a CMake script fragment is generated at the root of the cache-folder. In this manner that script would define version but could also contain copies for functions for how-to read the cache. With this approach, assuming the set of functions is well-defined, an old CPM would still utilize the cache-specific functions. This could also open some unexpected possibilities to support custom caching mechanisms that we couldn't yet imagine!

**Comment by CraigHutchinson** [comment]
I have discovered that setting `CPM_USE_NAMED_CACHE_DIRECTORIES` after a cache is already present does not incur any issues. This is also mentioned in the main `readme.md`  https://github.com/cpm-cmake/CPM.cmake/blob/a02a9601da8fa8cdf19cb5c9d0ae0c846443d595/README.md?plain=1#LL212C184-L213C51

**Comment by TheLartians** [comment]
That's true, the only real issue I see is maybe having too long path names due to the duplicate package name. I wouldn't be against making it default as long as we provide an option to opt-out in case there are issues in the first versions.
